cockatrice/src/settingscache.cpp

```diff
@@ -492,32 +493,32 @@ void SettingsCache::setMinPlayersForMultiColumnLayout(int _minPlayersForMultiCol
 
 void SettingsCache::setTapAnimation(int _tapAnimation)
 {
-    tapAnimation = _tapAnimation;
+    tapAnimation = static_cast<bool>(_tapAnimation);
     settings->setValue("cards/tapanimation", tapAnimation);
 }
 
 void SettingsCache::setChatMention(int _chatMention)
 {
-    chatMention = _chatMention;
+    chatMention = static_cast<bool>(_chatMention);
     settings->setValue("chat/mention", chatMention);
 }
 
 void SettingsCache::setChatMentionCompleter(const int _enableMentionCompleter)
 {
-    chatMentionCompleter = _enableMentionCompleter;
+    chatMentionCompleter = (bool)_enableMentionCompleter;
```

Comment:
Why not `static_cast` here?

Reply:
Clion changed it to be like this. I'm assuming it has something to do with the const functionality and it might need a `const_cast` which is dangerous. C style probably avoids this. @ctrlaltca ?

Reply:
The member var `chatMentionCompleter` is a boolean, while the `_enableMentionCompleter` parameter is an integer. Clion just made the "int to bool" cast explicit; no const involved.

Reply:
Dynamic cast then? I thought our style was to not use the c-style paren casts any more in favor of the cpp casts